<a href="https://colab.research.google.com/github/Rajat-Sri/Python-Coding-Practice/blob/main/Python_Coding_Practice.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Basic Operations In Python

## Performing Arithmetic Operations using Python

Let's begin by using Python as a calculator. You can write and execute Python using a code cell within Jupyter.


In [1]:
2+3+9

14

In [2]:
99-73

26

In [3]:
34*-56.7

-1927.8000000000002

In [4]:
3018/14

215.57142857142858

In [5]:
100//7

14

In [6]:
5**3

125

Python supports the following arithmetic operators:

| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |


## Variables:
 While working with a programming language such as Python, information is stored in variables. You can think of variables as containers for storing data. The data stored within a variable is called its value.

In [8]:
a = 4
b = 5
c = a+b
c

9

 **Comments**: Comments and blank lines are ignored during execution, but they are useful for providing information to humans (including yourself) about what the code does. Comments can be inline (at the end of some code), on a separate line, or even span multiple lines.

Inline and single-line comments start with `#`, whereas multi-line comments begin and end with three quotes, i.e. `"""`. Here are some examples of code comments:

In [9]:
my_favorite_number = 1 # an inline comment

In [10]:
# This comment gets its own line
my_least_favorite_number = 3

In [11]:
"""This is a multi-line comment.
Write as little or as much as you'd like.

Comments are really helpful for people reading
your code, but try to keep them short & to-the-point.

Also, if you use good variable names, then your code is
often self explanatory, and you may not even need comments!
"""
a_neutral_number = 5

## Evaluating conditions using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers & variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

In [12]:
my_favorite_number = 1
my_least_favorite_number = 5
a_neutral_number = 3

In [13]:
# Equality check - True
my_favorite_number == 1

True

In [14]:
# Equality check - False
my_favorite_number == my_least_favorite_number

False

In [15]:
# Not equal check - True
my_favorite_number != a_neutral_number

True

In [16]:
# Not equal check - False
a_neutral_number != 3

False

In [17]:
# Greater than check - True
my_least_favorite_number > a_neutral_number

True

In [18]:
# Greater than check - False
my_favorite_number > my_least_favorite_number

False

In [19]:
# Less than check - False
my_least_favorite_number < my_favorite_number

False

In [20]:
# Greater than or equal check - False
my_favorite_number >= 3

False

In [21]:
# Less than or equal check - False
my_favorite_number + a_neutral_number <= 3

False

Just like arithmetic operations, the result of a comparison operation can also be stored in a variable.

In [22]:
cost_of_ice_bag = 1.25
is_ice_bag_expensive = cost_of_ice_bag >= 10
print("Is the ice bag expensive?", is_ice_bag_expensive)

Is the ice bag expensive? False


## Combining conditions with logical operators

The logical operators `and`, `or` and `not` operate upon conditions and `True` & `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on a single condition.

The `and` operator returns `True` when both the conditions evaluate to `True`. Otherwise, it returns `False`.

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |

In [23]:
my_favorite_number

1

In [24]:
my_favorite_number > 0 and my_favorite_number <= 3

True

In [25]:
my_favorite_number < 0 and my_favorite_number <= 3

False

In [26]:
True and False

False

In [27]:
True and True

True

The `or` operator returns `True` if at least one of the conditions evaluates to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |


In [28]:
a_neutral_number = 3

In [29]:
a_neutral_number == 3 or my_favorite_number < 0

True

In [30]:
a_neutral_number != 3 or my_favorite_number < 0

False

In [31]:
my_favorite_number < 0 or True

True

The `not` operator returns `False` if a condition is `True` and `True` if the condition is `False`.

In [32]:
not a_neutral_number == 3

False

In [33]:
not my_favorite_number < 0

True

Logical operators can be combined to form complex conditions. Use round brackets or parentheses `(` and `)` to indicate the order in which logical operators should be applied.

In [34]:
(2 > 3 and 4 <= 5) or not (my_favorite_number < 0 and True)

True

In [35]:
not (True and 0 < 1) or (False and True)

False

If parentheses are not used, logical operators are applied from left to right.

In [36]:
not True and 0 < 1 or False and True

False

#Next Heading